sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotPassSameValueAsMultipleArguments.cs

```diff
@@ -0,0 +1,106 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class DoNotPassSameValueAsMultipleArguments : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S4142";
+        private const string MessageFormat =
+            "Verify that this is the intended value; it is the same as the {0} argument.";
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var invocation = (InvocationExpressionSyntax)c.Node;
+
+                if (invocation.ArgumentList?.Arguments.Count < 2)
+                {
+                    return;
+                }
+
+                var arguments = invocation.ArgumentList.Arguments;
+                var argsConvertedType = arguments
+                    .Select(arg => c.SemanticModel.GetTypeInfo(arg.Expression).ConvertedType)
+                    .ToList();
+
+                for (int i = 1; i < arguments.Count; i++)
+                {
+                    var parameterUsage = new HashSet<ISymbol> { argsConvertedType[i] };
+
+                    for (int j = 0; j < i; j++)
+                    {
+                        if (!IsLiteral(arguments[i]) &&
+                            arguments[i].IsEquivalentTo(arguments[j]) &&
+                            IsDuplicateUsage(parameterUsage, argsConvertedType[j]))
+                        {
+                            c.ReportDiagnostic(Diagnostic.Create(rule,
+                                arguments[i].GetLocation(),
+                                additionalLocations: new[] { arguments[j].GetLocation() },
+                                messageArgs: ToOrdinalNumberString(j + 1)));
+                            break;
```

Comment:
Should we report only once per method? Not a big deal, though.

Reply:
I'd say we should report multiple times, as there are separate issues. It could be that there are two issues, one of which is noise and second one a real problem.
Another argument IMHO is that would make for weird user experience - user would fix the problem only to see it is still there.